### ISSUE TITLE
Keras bidirectional lstm incorrect predictions

### ISSUE THREAD

**Issue by teaglin** [tensorflow/tensorflow#48858]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution Linux Ubuntu 20.04):
- TensorFlow installed from (source or binary): Binary & Source
- TensorFlow version (use command below): 2.4 & 2.3.1
- Python version: 3.8
- Bazel version (if compiling from source): 3.1
- CUDA/cuDNN version: 11 / 8
- GPU model and memory: RTX Quadro 8000 / 48GB

You can also obtain the TensorFlow version with:
I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.11.0
unknown 2.3.1

**Describe the current behavior**

I have a previous model written with the old Keras API before it was integrated into TensorFlow. We've been using this model, code and build setup for 2 years now no issues whatsoever. We decided to finally upgrade to TensorFlow 2. After upgrading we tried training. 

During training the model achieves the expected validation accuracy, but if we test the model after training it just spits out garbage values. For a sanity check we tested the model trained with the old version of Keras in TensorFlow 2 and it produces the correct output. Absolutely nothing has changed with the data or process. Only difference is the version of Keras and TensorFlow. We are stumped as to what the issue could be. The model is very simple, stacked bi-directional LSTMs. It also achieves the expected validation accuracy in training. 

Interestingly, this model is destined for CoreML, but exporting the model trained in TensorFlow 2 generates a larger model with significantly more layers than the previous. 

**Describe the expected behavior**

The trained model should produce the correct output.

**Standalone code to reproduce the issue**

This is the model.

`

        x = Input(shape=(length , feature_size,))
        branch_1 = Bidirectional(LSTM(256, return_sequences=True))(x)
        branch_1 = Bidirectional(LSTM(128, return_sequences=True))(branch_1)
        branch_1 = Bidirectional(LSTM(64, return_sequences=True))(branch_1)
        sequence = TimeDistributed(Dense(10, activation='softmax'), name='out')(branch_1)
        temporal_model = Model([x], [sequence])`

**Comment by UsharaniPagadala** [comment]
@teaglin 

In order to reproduce the issue reported here, could you please provide the complete code/colab gist along with the dependencies and the dataset you are using.Thanks!

**Comment by teaglin** [comment]
Looks like it was a hardware failure issue. Closing issue.
